src/orion/core/worker/storage_backup.py

```diff
@@ -0,0 +1,473 @@
+):
+    """Execute import
+
+    Parameters
+    ----------
+    dst_storage: BaseStorageProtocol
+        destination storage where to apply changes
+    queries_to_delete: dict
+        dictionary mapping a collection name to a list of queries to use
+        to find and delete data
+    data_to_add: dict
+        dictionary mapping a collection name to a list of data to add
+    progress_callback:
+        See :func:`load_database`
+    """
+
+    total_queries = sum(len(queries) for queries in queries_to_delete.values())
+    for collection_name in COLLECTIONS:
+        queries_to_delete.setdefault(collection_name, ())
+    i_query = 0
+    for query_delete_benchmark in queries_to_delete[COL_BENCHMARKS]:
+        logger.info(
+            f"Deleting from {len(queries_to_delete[COL_BENCHMARKS])} queries into {COL_BENCHMARKS}"
+        )
+        dst_storage.delete_benchmark(query_delete_benchmark)
+        _describe_import_progress(
+            STEP_DELETE_OLD_DATA, i_query, total_queries, progress_callback
+        )
+        i_query += 1
+    for query_delete_experiment in queries_to_delete[COL_EXPERIMENTS]:
+        logger.info(
+            f"Deleting from {len(queries_to_delete[COL_EXPERIMENTS])} queries "
+            f"into {COL_EXPERIMENTS}"
+        )
+        dst_storage.delete_experiment(uid=query_delete_experiment["_id"])
+        _describe_import_progress(
+            STEP_DELETE_OLD_DATA, i_query, total_queries, progress_callback
+        )
+        i_query += 1
+    for query_delete_trials in queries_to_delete[COL_TRIALS]:
+        logger.info(
+            f"Deleting from {len(queries_to_delete[COL_TRIALS])} queries into {COL_TRIALS}"
+        )
+        dst_storage.delete_trials(uid=query_delete_trials["experiment"])
+        _describe_import_progress(
+            STEP_DELETE_OLD_DATA, i_query, total_queries, progress_callback
+        )
+        i_query += 1
+    for query_delete_algo in queries_to_delete[COL_ALGOS]:
+        logger.info(
+            f"Deleting from {len(queries_to_delete[COL_ALGOS])} queries into {COL_ALGOS}"
+        )
+        dst_storage.delete_algorithm_lock(uid=query_delete_algo["experiment"])
+        _describe_import_progress(
+            STEP_DELETE_OLD_DATA, i_query, total_queries, progress_callback
+        )
+        i_query += 1
+
+    _describe_import_progress(
+        STEP_DELETE_OLD_DATA, total_queries, total_queries, progress_callback
+    )
+
+    nb_data_to_add = len(data_to_add.get(COL_BENCHMARKS, ())) + len(
+        data_to_add.get(COL_EXPERIMENTS, ())
+    )
+    i_data = 0
+
+    for new_benchmark in data_to_add.get(COL_BENCHMARKS, ()):
+        dst_storage.create_benchmark(new_benchmark)
+        _describe_import_progress(
+            STEP_INSERT_NEW_DATA, i_data, nb_data_to_add, progress_callback
+        )
+        i_data += 1
+
+    for new_experiment in data_to_add.get(COL_EXPERIMENTS, ()):
+        new_algo = data_to_add[COL_ALGOS][new_experiment["_id"]]
```

Comment:
So, in order to handle any type of IDs, I think a solution would be to generate a random hashing for the new experiment id as a common reference for the algo and trial documents (during `_prepare_import`). Then here, you could fetch the algo and trial documents just as you do here, but then pop the `_id` out of `new_experiments`, call `create_experiment` and then reassign the freshly created `new_experiment['_id']` by the storage backend to algo and trials documents.